### PR TITLE
MON-3890: add recording rule acm_capacity_effective_cpu_cores

### PR DESCRIPTION
**Pull request by elgnay**
The related issue: https://issues.redhat.com/browse/MON-3890

### CODE REVIEW

**Comment by simonpasquier** [comment]
/hold
depends on https://github.com/openshift/cluster-monitoring-operator/pull/2367

**Review by kahowell — APPROVED** [review]
LGTM

File: jsonnet/telemeter/rules.libsonnet [hunk 1]
```diff
@@ -185,6 +185,24 @@
...
+            {
+              // ACM managed cluster effective cores for subscription usage. It is measured in virtual CPU cores.
+              // 1. The value for the self managed OpenShift clusters is derived from the OpenShift metric cluster:capacity_effective_cpu_cores and is normalized to virtual CPU cores (* 2);
+              // 2. The value for the managed OpenShift clusters and the non-OpenShift clusters comes from the ACM metric acm_managed_cluster_worker_cores:max;
+              record: 'acm_capacity_effective_cpu_cores',
+              expr: |||
+                # self managed OpenShift cluster
+                max by (_id, managed_cluster_id) (
+                  acm_managed_cluster_info{product="OpenShift"} * on(managed_cluster_id) group_left() (
+                    label_replace(
```
Comment (simonpasquier) [inline jsonnet/telemeter/rules.libsonnet:197]: (suggestion) explain why the _id label is copied to the managed_cluster_id label
```suggestion
                    # On one side, the acm_managed_cluster_info metric has the managed_cluster_id label identifiying the spoke cluster and the _id label identifying the hub cluster.
                    # On the other side, the cluster:capacity_effective_cpu_cores metric has the _id label which identifying the spoke cluster.
                    # To join the 2 metrics, we need to add a managed_cluster_id label with the same value as _id to the cluster:capacity_effective_cpu_cores metric.
                    label_replace(
```

File: jsonnet/telemeter/rules.libsonnet [hunk 2]
```diff
@@ -185,6 +185,24 @@
...
+                      cluster:capacity_effective_cpu_cores, "managed_cluster_id", "$1", "_id", "(.*)"
```
Comment (simonpasquier) [inline jsonnet/telemeter/rules.libsonnet:198]: same suggestion regarding many-to-many
```suggestion
                      max by(_id) (cluster:capacity_effective_cpu_cores), "managed_cluster_id", "$1", "_id", "(.*)"
```

File: jsonnet/telemeter/rules.libsonnet [hunk 3]
```diff
@@ -185,6 +185,24 @@
...
             },
+            {
+              // ACM managed cluster effective cores for subscription usage. It is measured in virtual CPU cores.
+              // 1. The value for the self managed OpenShift clusters is derived from the OpenShift metric cluster:capacity_effective_cpu_cores and is normalized to virtual CPU cores (* 2);
+              // 2. The value for the managed OpenShift clusters and the non-OpenShift clusters comes from the ACM metric acm_managed_cluster_worker_cores:max;
+              record: 'acm_capacity_effective_cpu_cores',
+              expr: |||
+                # self managed OpenShift cluster
+                max by (_id, managed_cluster_id) (
+                  acm_managed_cluster_info{product="OpenShift"} * on(managed_cluster_id) group_left() (
```
Comment (simonpasquier) [inline jsonnet/telemeter/rules.libsonnet:196]: (suggestion) to exclude any possibility of many-to-many match errors.
```suggestion
                (
                  max by (_id, managed_cluster_id) (acm_managed_cluster_info{product="OpenShift"}) * on(managed_cluster_id) group_left() (
```

File: jsonnet/telemeter/rules.libsonnet (hunk 1 above)
Comment (elgnay) [reply jsonnet/telemeter/rules.libsonnet:199]: Comments added.

File: jsonnet/telemeter/rules.libsonnet (hunk 2 above)
Comment (elgnay) [reply jsonnet/telemeter/rules.libsonnet]: Updated as suggested.

File: jsonnet/telemeter/rules.libsonnet (hunk 3 above)
Comment (elgnay) [reply jsonnet/telemeter/rules.libsonnet]: Updated as suggested.

**Comment by simonpasquier** [comment]
/lgtm
/approve
I'll let @kahowell unhold if happy with the current version.

**Pull request merged**